how-to/how-to-use-kata-containers-with-acrn.md

```diff
@@ -0,0 +1,126 @@
+
+## Install and Configure Docker
+
+Docker 18.06 is required for running Kata with ACRN. For Kata+ACRN, a block-based driver like device-mapper is required. The latest release of Docker, 18.09, does not support device-mapper and is not compatible.
+
+To configure Docker for device-mapper and Kata, first, stop Docker daemon if it is already running using,
+
+```bash
+sudo systemctl stop docker
+```
+
+Set /etc/docker/daemon.json with the following contents:
+
+```bash
+{
+  "storage-driver": "devicemapper"
+}
+```
+
+Restart docker:
+
+```bash
+sudo systemctl daemon-reload
+sudo systemctl restart docker
+```
+
+Configure [Docker](https://github.com/kata-containers/documentation/blob/master/Developer-Guide.md#update-the-docker-systemd-unit-file) to use kata-runtime
+
+## Configure Kata Containers with ACRN
+
+To configure Kata Containers with ACRN, all that is needed is to copy the configuration-acrn.toml file (please find attached) that is generated when building the kata-runtime to either /etc/kata-containers/configuration.toml or /usr/share/defaults/kata-containers/configuration.toml.
+
+Below command shows full paths to the configuration.toml files that the runtime loads.  The first path that exists is used. (Please make sure the kernel and image paths are set correctly in the configuration.toml file)
+
+```bash
+sudo kata-runtime --kata-show-default-config-paths
+```
+
+Once the configuration.toml is updated, please __offline the CPUs__ by creating a bash script using below and executing it to offline the CPUs.
+
+```bash
+echo "dm_run: before offline cpu" > /dev/kmsg
+# offline SOS CPUs except BSP before launch UOS
+for i in `ls -d /sys/devices/system/cpu/cpu[1-99]`; do
+        online=`cat $i/online`
+        idx=`echo $i | tr -cd "[1-99]"`
+        echo cpu$idx online=$online
+        if [ "$online" = "1" ]; then
+                echo 0 > $i/online
+                online=`cat $i/online`
+                # during boot time, cpu hotplug may be disabled
+                # by pci_device_probe during a pci module insmod
+                while [ "$online" = "1" ]; do
+                        sleep 0.1
+                        echo 0 > $i/online
+                        online=`cat $i/online`
+                done
+                echo $idx > /sys/class/vhm/acrn_vhm/offline_cpu
+        fi
+done
+echo "dm_run: after offline cpu" > /dev/kmsg
+```
+
+Start ACRN based kata container,
+
+```bash
+sudo docker run -ti --runtime=kata-runtime busybox sh
+```
+
+You'll see ACRN is now running on your system, as well as a kata-shim and kata-proxy processes: (make sure the shell launched is running, on exiting all the processes are terminated)
```

Comment:
`kata-proxy`, `kata-shim`.

Reply:
*(you should obtain an interactive shell prompt; when exiting the container, make sure that all the Kata processes are terminated)*

Reply:
done